app/models/data_set.rb

```diff
@@ -65,8 +65,12 @@ def places_for_postcode(postcode, distance = nil, limit = nil)
       end
     else
       location_data = MapitApi.location_for_postcode(postcode)
-      location = Point.new(latitude: location_data.lat, longitude: location_data.lon)
-      places_near(location, distance, limit)
+      if location_data.lat && location_data.lon
```

Comment:
It actually fails here: https://github.com/alphagov/imminence/blob/master/app/models/point.rb#L13-L15

When `Float` is called on the nil lat/lon. So it needed to be caught before `places_near`. There was also a similar fall back for local authority snacs so it seemed best to have both in the same place.